src/cdk/explorer/nodes/constructNode.ts

```diff
@@ -7,32 +7,35 @@ import * as vscode from 'vscode'
 import { AWSTreeNodeBase } from '../../../shared/treeview/nodes/awsTreeNodeBase'
 import { cdk } from '../../globals'
 import * as treeInspector from '../tree/treeInspector'
-import { ConstructTreeEntity } from '../tree/types'
+import { ConstructProps, ConstructTreeEntity } from '../tree/types'
+import { generatePropertyNodes, PropertyNode } from './propertyNode'
 
 /**
  * Represents a CDK construct
  */
 export class ConstructNode extends AWSTreeNodeBase {
     private readonly type: string
+    private readonly properties: ConstructProps | undefined
 
     get tooltip(): string {
         return this.type || this.construct.path
     }
 
-    // TODO add tree path so we can guarantee id values are unique
     get id(): string {
-        return this.construct.path
+        return `${this.treePath}/${this.construct.path}`
     }
 
     public constructor(
         public readonly label: string,
         public readonly collapsibleState: vscode.TreeItemCollapsibleState,
-        public readonly construct: ConstructTreeEntity
+        public readonly construct: ConstructTreeEntity,
+        public readonly treePath: string
```

Comment:
It looks like `treePath` isn't used externally (so it could be private). It also looks like it is only used for the TreeItem.id property. What do you think about generalizing this, so that `ConstructNode` takes `parentNode: AwsTreeNodeBase` instead, and you can use parentNode.id when producing this node's id.

Reply:
I like it, making the change now.